### PR TITLE
Implement user stories US01 and US28, and create tests

### PR DESCRIPTION
**Pull request by codankra**
Implement user stories US01 and US28, and create tests

### CODE REVIEW

**Comment by codankra** [comment]
I understand @matteobart re-factored to include line numbers in errors by running util functions through the check() method. This only works during real gedcom file parsing (i.e. running main.py with a file). Should output from tests.py also show line numbers?

**Comment by matteobart** [comment]
Nope you don’t need to worry about it in the tests.py file. We don’t even have files/line numbers there. We are simply just creating the families/people and testing the functions. There is a world where that is what he expects, but given our implementation it would not make sense. 
TL;DR you should be fine ... I haven’t looked at your code yet tho

**Comment by matteobart** [comment]
It looks like everything is pretty solid, the one thing I see is that maybe reject illegitimate dates should call your date function rather than call them one after another each time. Everything else I see are micro issues.
